lib/extract-templates.js

```diff
@@ -0,0 +1,177 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+import * as util from 'ember-template-imports/src/util.js';
+
+export const SUPPORTED_EXTENSIONS = ['.js', '.ts', '.gjs', '.gts'];
+const LOCATION_START = Object.freeze({ line: 0, column: 0 });
+/**
+ * Processes results and corrects for template location offsets.
+ *
+ * @typedef {object} TemplateInfo
+ * @property {number} line
+ * @property {number} column
+ * @property {string} template
+ * @property {boolean} isEmbedded
+ *
+ * @param {string} moduleSource
+ * @param {string} relativePath
+ *
+ * @returns {TemplateInfo[]}
+ */
+export function extractTemplates(moduleSource, relativePath) {
+  // If no relativePath is present, assuming we might have templates.
+  let mightHaveTemplates = relativePath ? isSupportedScriptFileExtension(relativePath) : true;
+
+  if (!mightHaveTemplates) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
+
+  let parsed = parseTemplates(moduleSource, relativePath, {
+    templateTag: util.TEMPLATE_TAG_NAME,
+    templateLiteral: [
+      {
+        importPath: 'ember-cli-htmlbars',
+        importIdentifier: 'hbs',
+      },
+      {
+        importPath: '@ember/template-compilation',
+        importIdentifier: 'hbs',
+      },
+      {
+        importPath: util.TEMPLATE_LITERAL_MODULE_SPECIFIER,
+        importIdentifier: util.TEMPLATE_LITERAL_IDENTIFIER,
+      },
+      {
+        importPath: 'ember-cli-htmlbars-inline-precompile',
+        importIdentifier: 'default',
+      },
+      {
+        importPath: 'htmlbars-inline-precompile',
+        importIdentifier: 'default',
+      },
+      {
+        importPath: '@ember/template-compilation',
+        importIdentifier: 'precompileTemplate',
+      },
+    ],
+  });
+
+  // If after parsing we found no templates and we had no relative path,
+  // then we assume we had a hbs file in input.
+  if (parsed.length === 0 && !relativePath) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
+
+  let result = parsed.map((templateInfo) => {
+    let { start, end } = templateInfo;
+    let templateStart = start.index + start[0].length;
+
+    let template = moduleSource.slice(templateStart, end.index);
+
+    return makeTemplateInfo(
+      coordinatesOf(moduleSource, templateStart, end.index),
+      template,
+      templateInfo,
+      true
+    );
+  });
+
+  return result;
+}
+
+/**
+ * @param {object} location
+ * @param {number} location.line
+ * @param {number} location.column
+ * @param {string} template
+ * @param {object} templateInfo
+ * @param {boolean} isEmbedded
+ *
+ * @returns {TemplateInfo}
+ */
+function makeTemplateInfo({ line, column }, template, templateInfo, isEmbedded) {
+  return {
+    line,
+    column,
+    template,
+    isEmbedded,
+    isStrictMode: !isEmbedded || isStrictMode(templateInfo),
```

Comment:
`isStrictMode` is basically determined by which import path you used (or if it came from `<template>`), from the list of imports that are in the PR at the moment the following are strict:

* `import { hbs } from 'ember-template-imports';`
* `<template></template>`

Reply:
@rwjblue I updated `isStrictMode` based on your comments. Right now I'm checking that `importIdentifier === 'hbs' && importPath === 'ember-template-imports'`.
This becomes an issue when the identifier is aliased as in

```import { hbs as theHbs } from 'ember-template-imports'; ```

because `parseTemplates` will return `{ importIdentifier: 'theHbs', importPath: 'ember-template-imports' }` which of course breaks the above check. Maybe we should return the un-aliased version from `parseTemplates` or remove the `importIdentifier === 'hbs'` check from `isStrictMode`.
I added tests for these cases but we need to resolve this conversation before they can pass.